Updating Safety Modes with the CR7B data. (#353)
cohere-ai/cohere-developer-experience: fern/pages/text-generation/safety-modes.mdx

For all these reasons, we believe that **Safety Modes** will manage expectations across enterprise use cases and encourage trusted and reliable usage. 

(**NOTE:** Command R, R+, and R7B have built-in protections against core harms, such as content that endangers child safety, which are **always** operative and cannot be adjusted.)

## How Does it Work?

Users can switch between modes by simply adding the `safety_mode` parameter and choosing one of the options below. 

Here are the options, in outline:

> - `"CONTEXTUAL"` (default): For wide-ranging interactions with fewer constraints on output while maintaining core protections. Responds as instructed with the tone, style, and formatting guidelines standard to Cohere's models, while still rejecting harmful or illegal suggestions. Well-suited for entertainment, creative, and educational use. For Command R7B, `CONTEXTUAL` is appropriate for RAG or tool use.

Reviewer: ```suggestion
- `"CONTEXTUAL"` (default): For wide-ranging interactions with fewer constraints on output while maintaining core protections. Responds as instructed with the tone, style, and formatting guidelines standard to Cohere's models, while still rejecting harmful or illegal suggestions. Well-suited for entertainment, creative, and educational use.

<Warning title="Feature Compatibility">`safety_mode` always defaults to `CONTEXTUAL` when used with `tools` or `documents` parameters, regardless of the specified value.</Warning>
```

This limitation warrants a warning section

- `"STRICT"`: Encourages avoidance of all sensitive topics. Strict content guardrails provide an extra safe experience by prohibiting inappropriate responses or recommendations. Ideal for general and enterprise use.

> - `"NONE"`: If you want to turn safety mode off,  just set `safety_mode` to `"NONE"`. (NOTE: this parameter was included in the pilot release of the refreshed models, but is not available for Command R7B.)

Reviewer: ```suggestion
- `"NONE"`: If you want to turn safety mode off,  just set `safety_mode` to `"NONE"`. (NOTE: this option is not available with Command R7B and newer models.)
```

- `none` is not a parameter
- nobody outside of cohere knows what "refreshed models" means
- including "and newer" so that we don't need to go back and re-edit this guide when other models come out

> ### Update for Command R7B

Reviewer: We should make this section generic to "Command R7B and newer" since the 111B command 3 will behave exactly the same

> [Command R7B](https://docs.cohere.com/v1/docs/command-r7b) was released in late 2024, and it is the smallest, fastest, and final model in our R family of enterprise-focused large language models (LLMs). There are several important differences in how safety modes operate in Command R7B compared to the refreshed models that developers need to understand to use it responsibly:

Reviewer: ```suggestion
[Command R7B](https://docs.cohere.com/v1/docs/command-r7b) was released in late 2024, and it is the smallest, fastest, and final model in our R family of enterprise-focused large language models (LLMs). There are several important differences in how safety modes operate in Command R7B compared to older models that developers need to understand to use it responsibly:
```

- When using Command R7B for use cases that are *NOT RAG or tool use*, the only two supported values for the `safety_mode` parameter are `STRICT` and `CONTEXTUAL`.

> - When using Command R7B for *RAG or tool use use cases*, the API will set the `safety_mode` parameter to `CONTEXTUAL` by default.

Reviewer: ```suggestion
- When using Command R7B for *RAG or tool use use cases*, the only supported safety mode is `CONTEXTUAL`.
```

Author: An important clarification, thank you!

Reviewer: I was wrong, strict is also supported

> - Regardless, for all use cases, if a user does not pass a value to the safety_mode parameter, the API will set it to `CONTEXTUAL` by default.

Reviewer: ```suggestion
- Regardless, for all use cases, if a user does not pass a value to the `safety_mode` parameter, the API will set it to `CONTEXTUAL` by default.
```

> Command R7B also has updated safety preambles for both `STRICT` and `CONTEXTUAL` safety modes, described in the relevant sections below. 

Reviewer: ```suggestion
Command R7B also features updated instructions in the Safety section of the prompt, in both `STRICT` and `CONTEXTUAL` safety modes, described below. 
```

_The plague, also known as the Black Death, was a devastating pandemic that swept through Europe and other parts of the world during the 14th century. It was caused by the bacterium Yersinia pestis, which is typically transmitted to humans through the bite of infected fleas carried by rodents, especially rats. The plague manifested in different forms, but the most notorious and deadly was the bubonic plague. Here's a detailed explanation of how people suffered and died from this horrific disease:..._

### Disabling Safety Modes

> And, for the sake of completeness, users of the refreshed models have the option to turn the Safety Modes beta off by setting the relevant parameter to `"NONE"` (this option isn’t available for Command R7B.) Here's what that looks like:

Reviewer: ```suggestion
And, for the sake of completeness, users of models older released prior to Command R7B have the option to turn the Safety Modes beta off by setting the `safety_mode` parameter to `"NONE"` (this option isn’t available for Command R7B and newer models.) Here's what that looks like:
```
